src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/AS_StrandBiasTest.java

```diff
@@ -0,0 +1,229 @@
+    }
+
+    @Override
+    public String getRawKeyName() { return GATKVCFConstants.AS_SB_TABLE_KEY; }
+
+    @Override
+    public Map<String, Object> annotate(final ReferenceContext ref,
+                                        final VariantContext vc,
+                                        final Map<String, PerReadAlleleLikelihoodMap> stratifiedPerReadAlleleLikelihoodMap) {
+        return annotateRawData(ref, vc, stratifiedPerReadAlleleLikelihoodMap);
+    }
+
+    @Override
+    public Map<String, Object> annotateRawData(final ReferenceContext ref,
+                                               final VariantContext vc,
+                                               final Map<String, PerReadAlleleLikelihoodMap> perReadAlleleLikelihoodMap ) {
+
+        //for allele-specific annotations we only call from HC and we only use perReadAlleleLikelihoodMap
+        if ( perReadAlleleLikelihoodMap == null) {
+            return new HashMap<>();
+            // calculate the annotation from the stratified per read likelihood map
+            // stratifiedPerReadAllelelikelihoodMap can come from HaplotypeCaller call to VariantAnnotatorEngine
+        } else if (perReadAlleleLikelihoodMap != null) {
+            final Map<String, Object> annotations = new HashMap<>();
+            final ReducibleAnnotationData<List<Integer>> myData = new AlleleSpecificAnnotationData<>(vc.getAlleles(),null);
+            calculateRawData(vc, perReadAlleleLikelihoodMap, myData);
+            final Map<Allele, List<Integer>> perAlleleValues = myData.getAttributeMap();
+            final String annotationString = makeRawAnnotationString(vc.getAlleles(), perAlleleValues);
+            annotations.put(getRawKeyName(), annotationString);
+            return annotations;
+        } else {
+            // for non-snp variants, we need per-read likelihoods.
+            // for snps, we can get same result from simple pileup
+            // for indels that do not have a computed strand bias (SB) or strand bias by sample (SBBS)
+            return null;
+        }
+    }
+
+    protected String makeRawAnnotationString(final List<Allele> vcAlleles, final Map<Allele, List<Integer>> perAlleleValues) {
+        String annotationString = "";
+        for (final Allele a : vcAlleles) {
+            if (!annotationString.isEmpty())
+                annotationString += printDelim;
+            List<Integer> alleleValues = perAlleleValues.get(a);
+            if (alleleValues == null)
+                alleleValues = ZERO_LIST;
+            annotationString += encode(alleleValues);
+        }
+        return annotationString;
+    }
+
+    protected String encode(List<Integer> alleleValues) {
+        String annotationString = "";
+        for (int j =0; j < alleleValues.size(); j++) {
+            annotationString += alleleValues.get(j);
+            if (j < alleleValues.size()-1)
+                annotationString += ",";
```

Comment:
\[FindBugs\] INFO: SBSC: org\.broadinstitute\.hellbender\.tools\.walkers\.annotator\.AS\_StrandBiasTest\.encode\(List\) concatenates strings using \+ in a loop